src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -31,6 +32,16 @@ const int64_t kUnhandledErrorGracePeriodNanos = static_cast<int64_t>(5e9);
 // when there are too many local objects.
 const int kMaxUnhandledErrorScanItems = 1000;
 
+namespace {
+
+Status signal_status = Status::OK();
```

Comment:
this has the possibility of being called from multiple threads right, should we make this safe by making it thread_local or guarded by a mutex?